### PR TITLE
[ao][sparsity][fx] make quant prepare -> sparse prepare compose

### PR DESCRIPTION
**Pull request by HDCharles**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82204
* #81994
* #81993
* __->__ #81992

Summary: sparse_prepare automatically composes with quantized prepare
even in cases with fusion. However, the convert step needed to be updated to handle parametrized
modules.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: test/ao/sparsity/test_composability.py [hunk 1]
```diff
@@ -312,5 +311,68 @@ def test_qat_prep_before_s_prep(self):
...
+        ) = _get_model_and_sparsifier_and_sparse_config()
+
+        example = torch.randn(1, 4, 4, 4)
+        qconfig = tq.get_default_qconfig("fbgemm")
+        qconfig_mapping = tq.QConfigMapping() \
+            .set_module_name("4", qconfig) \
+            .set_module_name("5", qconfig)
+
+
+        mod = prepare_fx(mod, qconfig_mapping, (example))
```
Comment (jerryzh168) [inline test/ao/sparsity/test_composability.py:340]: nit: example_inputs is a Tuple[Any, ...], so maybe `(example)` --> `(example,)`

**Review by jerryzh168 — APPROVED** [review]
looks good

File: test/ao/sparsity/test_composability.py [hunk 2]
```diff
@@ -312,5 +311,68 @@ def test_qat_prep_before_s_prep(self):
...
+def _module_has_activation_post_process(model, fqn_of_module):
+    for node in model.graph.nodes:
+        # look for an observer whose arg is the target module
+        if "activation_post_process" in node.name:
+            if node.args[0].target == fqn_of_module:
+                return True
+    return False
+
+class TestFxComposability(TestCase):
+    def test_q_prep_fx_before_s_prep(self):
```
Comment (jerryzh168) [inline test/ao/sparsity/test_composability.py:326]: can we add some comments for this test describing what this is trying to test

Comment (HDCharles) [reply test/ao/sparsity/test_composability.py:326]: yeah, will add more details to all the tests

File: test/ao/sparsity/test_composability.py (hunk 1 above)
Comment (HDCharles) [reply test/ao/sparsity/test_composability.py:340]: thanks, will fix

**Pull request closed without merging**